itensor/tagset.h

```diff
@@ -0,0 +1,228 @@
+//
+// Distributed under the ITensor Library License, Version 1.2
+//    (See accompanying LICENSE file.)
+//
+#ifndef __ITENSOR_TAGSET_H
+#define __ITENSOR_TAGSET_H
+#include "itensor/smallstring.h"
+
+namespace itensor {
+
+class TagSet;
+
+//
+// TagSet
+//
+
+class TagSet
+    {
+    public:
+    using tags_type = std::array<Tag,MAX_TAGS>;
+    private:
+    tags_type tags_;
+    size_t size_ = 0;
+    public:
+
+    TagSet() {};
+
+    //
+    // Convert a character array to a TagSet.
+    // Parses the array for ',' and splits up into tags, for example:
+    // "a,b,c" -> Tag("a"),Tag("b"),Tag("c")
+    TagSet(const char* t);
+
+    std::string
+    to_string()
```

Comment:
Usually I've been naming class methods using "camel case" so I'd use toString instead of to_string. Was the choice of to_string to match another existing convention though? I think using c_str below is good, in contrast, because it is parallel to the similar method that std::string defines. (So ok to break the rules sometimes, especially for advanced/developer functionality.)

Reply:
There was no particular reason for the name, I agree we should be consistent about using camel-casing.